### ISSUE TITLE
Overfitting problem

### ISSUE THREAD

**Issue by amyburden** [javier2491/FaceQnet#15]
Thanks so much for providing reference image list. I try to train on those 300 ids. I found overfitting problems during training. Model gets very low loss on training set while give similar outputs score for validation data. Have you ever encountered such problem. I did not do any augmentation on training image. Using dropout at first layer, wd 0.0004

**Comment by wb315** [comment]
@amyburden  hi, i generate the 300ids quality, but i found the quality is almost very small, do you have this problems?

**Comment by wb315** [comment]
hi, how can you generate the train dataset label? i train on those 300 ids, but the model is underfitting.

**Comment by amyburden** [comment]
Use pre-trained model output as gt

> hi, how can you generate the train dataset label? i train on those 300 ids, but the model is underfitting.

**Comment by wb315** [comment]
> Use pre-trained model output as gt
> 
> > hi, how can you generate the train dataset label? i train on those 300 ids, but the model is underfitting.

hi, can you open your code, thanks.

**Comment by javier-hernandezo** [comment]
> Thanks so much for providing reference image list. I try to train on those 300 ids. I found overfitting problems during training. Model gets very low loss on training set while give similar outputs score for validation data. Have you ever encountered such problem. I did not do any augmentation on training image. Using dropout at first layer, wd 0.0004

Maybe your learning rate is to high? Other option is to increase the dropout rate, i think it is too low. 
When I try the model first times I found this kind of problems and "playing" with these two parameters solved it. 
If you give me more details about your implementation I couldbe able to help you more.

Regards

**Comment by amyburden** [comment]
> > Thanks so much for providing reference image list. I try to train on those 300 ids. I found overfitting problems during training. Model gets very low loss on training set while give similar outputs score for validation data. Have you ever encountered such problem. I did not do any augmentation on training image. Using dropout at first layer, wd 0.0004
> 
> Maybe your learning rate is to high? Other option is to increase the dropout rate, i think it is too low.
> When I try the model first times I found this kind of problems and "playing" with these two parameters solved it.
> If you give me more details about your implementation I couldbe able to help you more.
> 
> Regards

I use insightface face recognition model which is one of SOTA model. I only fine-tuned the last layer of the model with LR 1e-4 and Drop out rate 0.5. Final l2 loss is 0.03.

After that, I try to use distillation method, which improves the loss to 0.0045. And poor quality image have lower score. But score range from 0.4 to 0.7, which is still not as good as yours. During training I use all vgg2 image and use augmentation including motion blur, brightness. So I am wondering it may be caused by data inconsistency that my first model failed and data distribution that my second model have narrow range.

BTW, for first method, normalized distance seems ambiguous to me in your paper. For example, How did you pick the mean and variance? Did them keep across different identity? Did you first normalize feature and then calculate the distance? It seems that you only use vgg2 dataset in your training which distance range 0 to 0.5. How can model output 1.0 score with such imbalanced data? The model you provided really works. I am also impressed with the idea. I appreciate it if you can provide more details. I am really happy to hear from you.

**Comment by javier-hernandezo** [comment]
If with your second try (data distillation, etc) your model is learning properly, and if the problem is that the ouptut scores are only in the 0.4 - 0.7 range, I would look at the normalization of the training data. Maybe your training scores are not properly distributed in all the 0 - 1 range, with few training samples with scores close to 0 and 1. 

However, having final quality scores between 0.4 and 0.7 is not a problem if you know that is happening. You can transform the final scores to take them to the desired final range.

I think that the normalization of the training distances is one of the main points to improve, I make a simple implementation, and I am pretty sure that are better way to do it: 

I just take a portion of the training scores (distances without any normalization) of several users (I do not remember exactly how many users) and I calculate the mean and std of the distances using them. Then I normalize the rest of the training data and transform it to normalized scores. I do not have into account different identities.  You can try using more or less training data for obtaining the mean and std until you are satisfied with the result. That is what I did, and as I say, it has a ton of ways to be improved. 

"It seems that you only use vgg2 dataset in your training which distance range 0 to 0.5." : I do not know why are you considering that, but with the process I described you in this message, the output scores should be in the proper range.

"The model you provided really works. I am also impressed with the idea." : Thank you very much. I will try to help you as much as I can. I really like to see that FaceQnet is being used by other people.

Regards
